genesis/engine/sensors/camera.py

```diff
@@ -0,0 +1,936 @@
+        camera_wrapper = self._get_camera_wrapper()
+        camera_wrapper.transform = tensor_to_array(transform)
+        self._shared_metadata.renderer.update_camera(camera_wrapper)
+
+    def _get_camera_wrapper(self):
+        """Get (and lazily create) the persistent camera wrapper for the renderer."""
+
+        if self._camera_wrapper is None:
+            self._camera_wrapper = RasterizerCameraWrapper(self)
+
+        return self._camera_wrapper
+
+    def apply_camera_transform(self, camera_T: torch.Tensor):
+        """Update rasterizer camera wrapper from a world transform."""
+        camera_wrapper = self._get_camera_wrapper()
+        camera_wrapper.transform = tensor_to_array(camera_T)
+        self._shared_metadata.renderer.update_camera(camera_wrapper)
+
+    def render_current_state(self):
+        """Perform the actual render for the current state."""
+        # Update scene state
+        self._shared_metadata.context.update(force_render=False)
+
+        # Render this camera
+        rgb_arr, _, _, _ = self._shared_metadata.renderer.render_camera(
+            self._get_camera_wrapper(),
+            rgb=True,
+            depth=False,
+            segmentation=False,
+            normal=False,
+        )
+
+        # Convert to torch tensor and store in cache
+        rgb_tensor = torch.from_numpy(rgb_arr.copy()).to(dtype=torch.uint8, device=gs.device)
+
+        # Store in cache
+        # rgb_arr shape: (H, W, 3) for single env or needs batching
+        n_envs = self._manager._sim._B
+        if n_envs == 0:
+            # Single environment case - add batch dimension
+            self._shared_metadata.image_cache[self._idx][0] = rgb_tensor
+        else:
+            # TODO: Handle multi-env rendering properly
+            # For now, just replicate the single render
+            for i in range(n_envs):
+                self._shared_metadata.image_cache[self._idx][i] = rgb_tensor
+
+    def add_light(
+        self,
+        pos=(0.0, 0.0, 5.0),
+        dir=(0.0, 0.0, -1.0),
+        color=(1.0, 1.0, 1.0),
+        intensity=1.0,
+        type="directional",
+    ):
+        """
+        Add a light to the scene for all rasterizer cameras.
+
+        Parameters
+        ----------
+        pos : tuple[float, float, float]
+            Light position.
+        dir : tuple[float, float, float]
+            Light direction (for directional lights).
+        color : tuple[float, float, float]
+            Light color RGB.
+        intensity : float
+            Light intensity.
+        type : str
+            Light type: "directional" or "point".
+        """
+        if self._shared_metadata.lights is not None:
+            light_dict = {
+                "type": type,
+                "pos": pos,
+                "dir": dir,
+                "color": tuple(np.array(color) * intensity),
+                "intensity": intensity,
+            }
+            self._shared_metadata.lights.append(light_dict)
```

Comment:
Why are you adding lights options to shared metadata?

Reply:
lights are shared across camera sensors of the same rendering backend. It's not used here, we can remove it if preferred.